store/kv/measure.go

```diff
@@ -108,6 +166,9 @@ func (m *TxImplWithMetrics) Delete(ctx context.Context, table []byte, key Key) (
 		err = m.tx.Delete(ctx, table, key)
 		return err
 	})
+	// Delete reads all the records that are about to be deleted and those reads are counted in the iterator.
+	// This data will be counted as writes instead of reads.
+	countReadsAsWrites(ctx)
```

Comment:
If I issue a delete where name = X, and this results in reading 100 records but only 10 of them need to be deleted, how many write units will be counted here?

Reply:
The only special handling for delete is filter `{}`, for which we count 1 drop unit.

Reply:
Shouldn't this be counted as 90 reads and 10 writes though? Why are we converting reads to writes?

Reply:
I was implementing the count at the level of the kv interface, what you describe is doable, but only with counting at the query runner level, like we do with delete with no filter. I am changing this to work as you described.